src/Symfony/Component/Console/Color.php

```diff
@@ -0,0 +1,165 @@
+        'white' => 7,
+        'default' => 9,
+    ];
+
+    private static $availableOptions = [
+        'bold' => ['set' => 1, 'unset' => 22],
+        'underscore' => ['set' => 4, 'unset' => 24],
+        'blink' => ['set' => 5, 'unset' => 25],
+        'reverse' => ['set' => 7, 'unset' => 27],
+        'conceal' => ['set' => 8, 'unset' => 28],
+    ];
+
+    private $foreground;
+    private $background;
+    private $options = [];
+
+    public function __construct(string $foreground = '', string $background = '', array $options = [])
+    {
+        $this->foreground = $this->parseColor($foreground);
+        $this->background = $this->parseColor($background);
+
+        foreach ($options as $option) {
+            if (!isset(static::$availableOptions[$option])) {
+                throw new InvalidArgumentException(sprintf('Invalid option specified: "%s". Expected one of (%s).', $option, implode(', ', array_keys(static::$availableOptions))));
+            }
+
+            $this->options[$option] = static::$availableOptions[$option];
+        }
+    }
+
+    public function apply(string $text): string
+    {
+        return $this->set().$text.$this->unset();
+    }
+
+    public function set(): string
+    {
+        $setCodes = [];
+        if ('' !== $this->foreground) {
+            $setCodes[] = '3'.$this->foreground;
+        }
+        if ('' !== $this->background) {
+            $setCodes[] = '4'.$this->background;
+        }
+        foreach ($this->options as $option) {
+            $setCodes[] = $option['set'];
+        }
+        if (0 === \count($setCodes)) {
+            return '';
+        }
+
+        return sprintf("\033[%sm", implode(';', $setCodes));
+    }
+
+    public function unset(): string
+    {
+        $unsetCodes = [];
+        if ('' !== $this->foreground) {
+            $unsetCodes[] = 39;
+        }
+        if ('' !== $this->background) {
+            $unsetCodes[] = 49;
+        }
+        foreach ($this->options as $option) {
+            $unsetCodes[] = $option['unset'];
+        }
+        if (0 === \count($unsetCodes)) {
+            return '';
+        }
+
+        return sprintf("\033[%sm", implode(';', $unsetCodes));
+    }
+
+    private function parseColor(string $color): string
+    {
+        if ('' === $color) {
+            return '';
+        }
+
+        if ('#' === $color[0]) {
+            $color = substr($color, 1);
+
+            if (3 === \strlen($color)) {
+                $color = $color[0].$color[0].$color[1].$color[1].$color[2].$color[2];
+            }
+
+            if (6 !== \strlen($color)) {
+                throw new InvalidArgumentException(sprintf('Invalid "%s" color.', $color));
+            }
+
+            return $this->convertHexColorToAnsi(hexdec($color));
+        }
+
+        if (!isset(static::$colors[$color])) {
+            throw new InvalidArgumentException(sprintf('Invalid "%s" color; expected one of (%s).', $color, implode(', ', array_keys(static::$colors))));
```

Comment:
maybe we can point out here that the user should use one of the predefined colors or an hexadecimal one ?